services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/tormaystarkastelu/TormaystarkasteluLaskentaService.kt

```diff
@@ -132,15 +132,13 @@ class TormaystarkasteluLaskentaService(
 
     internal fun linjaautoliikenneluokittelu(geometriaIds: Set<Int>): Int {
         if (tormaysService.anyIntersectsCriticalBusRoutes(geometriaIds)) {
-            return Linjaautoliikenneluokittelu
-                .KAMPPI_RAUTATIENTORI_MANNERHEIMINTIE_KAISANIEMENKATU_HAMEENTIE_TAI_YLI_20_VUOROA_RUUHKATUNNISSA
-                .value
+            return Linjaautoliikenneluokittelu.TARKEIMMAT_JOUKKOLIIKENNEKADUT.value
         }
 
         val bussireitit = tormaysService.getIntersectingBusRoutes(geometriaIds)
 
         val valueByRunkolinja =
-            bussireitit.maxOfOrNull { it.runkolinja.toLinjaautoliikenneluokittelu().value }
+            bussireitit.maxOfOrNull { it.runkolinja?.toLinjaautoliikenneluokittelu()?.value ?: 0 }
```

Comment:
Can we safely assume zero is a good default value in case the value is somehow missing from the database? Would it be better to fail the operation if our database is so badly messed up?

Why was this change made in an unrelated "PR comment changes" commit?

Reply:
I was thinking of those "almost" values that possibly are in the database. At least I have them in my current database. Default value EI could be used for those cases.

Reply:
I think it would be better for the code to fail than to continue with possible faulty values.

If I understood this correctly, the only this can fail is if the bus data in the database is from before "almost" was dropped from the processing result, i.e. from before 9.11.2022. Or if someone has added bad data directly to the DB. So if this fails in an environment, we definitely should update the bus data in that environment.

Reply:
"almost" values are still present in at least dev and test environments:
```
haitatonhanketietokantatest=> select distinct trunk from tormays_buses_polys;
 trunk  
--------
 no
 yes
 almost
(3 rows)
```

Reply:
I will try to generate a new git package and see what kind of data it holds.

Reply:
Okay, new gpkg generated does not contain any "almost" rows. So, it needs to be updated as well.

Reply:
I made it throw an exception when there is other value than 'yes' or 'no'.